src/pages/studyView/charts/ChartContainer.tsx

```diff
@@ -91,49 +91,124 @@ export class ChartContainer extends React.Component<IChartContainerProps, {}> {
+    }
 
+    @computed get chartControls(): ChartControls {
+        let controls = {};
+        switch (this.props.chartMeta.chartType) {
+            case ChartType.PIE_CHART: {
+                controls = { showTableIcon: true }
+                break;
+            }
+            case ChartType.TABLE: {
+                if (!_.isEqual(this.props.chartMeta.defaultChartType, ChartType.TABLE)) {
+                    controls = { showPieIcon: true }
+                }
+                break;
+            }
+        }
+        return { ...controls, showResetIcon: this.chartFilters.length > 0 };
+    }
+
+    @bind
+    @action changeChartType(chartType:ChartType){
+        this.props.changeChartType(this.props.chartMeta.uniqueKey,chartType)
+    }
+
+    public render() {
         return (
-            <If condition={this.isChartVisible}>
-                <div className={styles.studyViewChart}
-                     onMouseEnter={this.handlers.onMouseEnterPlot}
-                     onMouseLeave={this.handlers.onMouseLeavePlot}>
-                    <ChartHeader
-                        clinicalAttribute={this.props.clinicalAttribute}
-                        showControls={this.mouseInPlot}
-                        showResetIcon={this.props.filters.length > 0}
-                        handleResetClick={this.handlers.resetFilters}
-                    />
-                    <div style={{flexGrow: 1, display: 'flex', alignItems:'center'}}>
-                        <StudyViewComponentLoader promise={this.props.promise}>
-                            <If condition={this.props.chartType === ChartType.PIE_CHART}>
-                                <PieChart
-                                    ref={this.handlers.ref}
-                                    onUserSelection={this.handlers.onUserSelection}
-                                    filters={this.props.filters}
-                                    data={this.data}/>
-                            </If>
-                        </StudyViewComponentLoader>
-                    </div>
-                </div>
-            </If>
+            <div className={ classnames(styles.chart, this.chartWidth, this.chartHeight) }
+                    onMouseEnter={this.handlers.onMouseEnterChart}
+                    onMouseLeave={this.handlers.onMouseLeaveChart}>
+                <ChartHeader 
+                    clinicalAttribute={this.props.chartMeta.clinicalAttribute}
+                    active={this.mouseInChart}
+                    resetChart={this.handlers.resetFilters}
+                    deleteChart={this.handlers.onDeleteChart}
+                    hideLabel={this.hideLabel}
+                    chartControls={this.chartControls}
+                    changeChartType={this.changeChartType}
+                />
+
+                <If condition={this.props.chartMeta.chartType === ChartType.PIE_CHART}>
+                    <StudyViewComponentLoader promise={this.data}>
```

Comment:
The if condition should be placed inside of StudyViewComponentLoader. This component will be used for all charts, then you can avoid duplication later.

Reply:
if I do that and try changing the chart type, pie to table or vice-versa, nothing shows up.